src/Microsoft.AspNetCore.Razor.Runtime/Hosting/RazorCompiledItemMetadataAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Hosting
+{
+    /// <summary>
+    /// Defines a key/value metadata pair for the decorated Razor type.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
```

Comment:
why inherited = false?

Reply:
Nearly all the attributes in Razor are marked `Inherited = false`. I thought you guys were on to something.

Reply:
The assembly attributes?

Reply:
Yup:

```
src\Microsoft.AspNetCore.Razor\TagHelpers\HtmlAttributeNameAttribute.cs:11:
[AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = false)]
src\Microsoft.AspNetCore.Razor\TagHelpers\HtmlAttributeNotBoundAttribute.cs:11:
[AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = false)]
src\Microsoft.AspNetCore.Razor\TagHelpers\HtmlTargetElementAttribute.cs:11:    [AttributeUsage(AttributeTargets.Class,
AllowMultiple = true, Inherited = false)]
src\Microsoft.AspNetCore.Razor\TagHelpers\OutputElementHintAttribute.cs:11:    [AttributeUsage(AttributeTargets.Class,
AllowMultiple = false, Inherited = false)]
src\Microsoft.AspNetCore.Razor\TagHelpers\RestrictChildrenAttribute.cs:15:    [AttributeUsage(AttributeTargets.Class,
Inherited = false, AllowMultiple = false)]
src\Microsoft.AspNetCore.Razor.Language\ProvideRazorExtensionInitializerAttribute.cs:8:
[AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true, Inherited = false)]
src\Microsoft.AspNetCore.Razor.Runtime\Hosting\RazorCompiledItemAttribute.cs:11:
[AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true, Inherited = false)]
src\Microsoft.AspNetCore.Razor.Runtime\Hosting\RazorCompiledItemMetadataAttribute.cs:11:
[AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
src\Microsoft.AspNetCore.Razor.Runtime\Hosting\RazorConfigurationNameAttribute.cs:16:
[AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false, Inherited = false)]
src\Microsoft.AspNetCore.Razor.Runtime\Hosting\RazorExtensionAssemblyNameAttribute.cs:16:
[AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true, Inherited = false)]
src\Microsoft.AspNetCore.Razor.Runtime\Hosting\RazorLanguageVersionAttribute.cs:16:
[AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false, Inherited = false)]
src\Microsoft.AspNetCore.Razor.Runtime\Hosting\RazorSourceChecksumAttribute.cs:21:
[AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = true)]
src\Microsoft.CodeAnalysis.Razor.Workspaces\ExportCustomProjectEngineFactoryAttribute.cs:10:
[AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
test\Microsoft.AspNetCore.Razor.Test.MvcShim\Microsoft.AspNetCore.Mvc.ApplicationParts\ProvideApplicationPartFactoryAtt
ribute.cs:8:    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false)]
test\Microsoft.AspNetCore.Razor.Test.MvcShim\Microsoft.AspNetCore.Mvc.ApplicationParts\RelatedAssemblyAttribute.cs:8:
  [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
test\Microsoft.AspNetCore.Razor.Test.MvcShim\Microsoft.AspNetCore.Mvc.Razor\RazorViewAttribute.cs:8:
[AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
test\Microsoft.VisualStudio.Editor.Razor.Test.Common\Xunit\ForegroundFactAttribute.cs:10:
[AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
test\Microsoft.VisualStudio.Editor.Razor.Test.Common\Xunit\ForegroundTheoryAttribute.cs:10:
[AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
```

Reply:
Lets let this be inherited

Reply:
This attribute is for classes though. That's it should be inherited=true